modules/light-clients/08-wasm/keeper/snapshotter.go

```diff
@@ -0,0 +1,115 @@
+package keeper
+
+import (
+	"io"
+
+	errorsmod "cosmossdk.io/errors"
+
+	snapshot "cosmossdk.io/store/snapshots/types"
+	storetypes "cosmossdk.io/store/types"
+	tmproto "github.com/cometbft/cometbft/proto/tendermint/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/ibc-go/modules/light-clients/08-wasm/types"
+)
+
+var _ snapshot.ExtensionSnapshotter = &WasmSnapshotter{}
+
+// SnapshotFormat format 1 is just gzipped wasm byte code for each item payload. No protobuf envelope, no metadata.
+const SnapshotFormat = 1
+
+type WasmSnapshotter struct {
+	wasm *Keeper
+	cms  storetypes.MultiStore
+}
+
+func NewWasmSnapshotter(cms storetypes.MultiStore, wasm *Keeper) *WasmSnapshotter {
```

Comment:
could we add godoc for all public funcs?

Reply:
addressed in https://github.com/cosmos/ibc-go/pull/4723/commits/43b71dbce89f29158a9017c62d6d2a72a84b6854